### PR TITLE
Add initial repo and code standards

### PR DESCRIPTION
**Pull request by manics**
Closes https://github.com/jupyterhub/team-compass/issues/454

This is the text suggested in https://github.com/jupyterhub/team-compass/issues/454#issuecomment-939319272 with some minor tweaks.

### CODE REVIEW

File: docs/team/repository-code-standards.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - the amount of code churn
+  - how it will improve code maintainability
+  - the time it may add to CI runs.
+
+## Configuration of a tool
+
+Use your judgment as to whether to apply a strict or lenient configuration for linting code.
+Example configurations are in
+  - JupyterHub (A large repository which was converted and therefore has a fairly lax configuration)
+  - **TODO: Repository where pre-commit/flake8 was enforced near the start so has a strict configuration**
```
Comment (consideRatio) [inline docs/team/repository-code-standards.md:29]: It isn't enforced near start, but it is quite strict currently. It didn't cause much trouble to make the change as there were no other open PRs to cause conflict against.

See https://github.com/jupyterhub/nativeauthenticator's [.pre-commit-config.yaml](https://github.com/jupyterhub/nativeauthenticator/blob/main/.pre-commit-config.yaml) and [.flake8](https://github.com/jupyterhub/nativeauthenticator/blob/main/.flake8) config.

**Review by consideRatio — APPROVED** [review]
Thanks @manics for nudging this towards resolution and thank you @willingc for writing this draft!

I think this is a good starting point!

File: docs/team/repository-code-standards.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Repository Management and Code Standards
+
+JupyterHub projects should take advantage of pre-commit and tools to help maintain consistent formatting within a repo to improve overall code quality, review efficiency, and readability of code.
+Sensible use of pre-commit to run formatting tools and code linters can add consistency and improve maintainability.
+
+## Preferred tools and pre-commit hooks
+
+The team has found the following tools and their pre-commit hooks to be useful. The following pre-commit hooks can be added to any repo when convenient, though please communicate with anyone who has an open pull request if it will lead to major conflicts:
+
+  - [black](https://black.readthedocs.io/)
```
Comment (choldgraf) [inline docs/team/repository-code-standards.md:10]: worth recommending to just use default settings for these whenever possible?

File: docs/team/repository-code-standards.md (hunk 1 above)
Comment (manics) [reply docs/team/repository-code-standards.md]: Added!

File: docs/team/repository-code-standards.md (hunk 2 above)
Comment (manics) [reply docs/team/repository-code-standards.md:10]: Added to the `Configuration of a tool` section

**Review by willingc — APPROVED** [review]
Thanks @manics.

**Pull request merged**